### PR TITLE
New Rule 61 - valid cookbook version

### PR DESCRIPTION
**Pull request by lamont-granquist**
closes #351

### CODE REVIEW

File: lib/foodcritic/rules.rb [hunk 1]
```diff
@@ -826,3 +826,28 @@ def invalid_name(ast)
...
+rule 'FC061', 'Valid cookbook version should be defined' do
+  tags %w{metadata}
+  metadata do |ast, filename|
+    ver = ast.xpath(%Q(//command[ident/@value='version']/descendant::tstring_content/@value))
+    if ver.nil? || ver == ''
+      [file_match(filename)]
+    else
+      pass = true
+      parts = ver.to_s.gsub(/\s|'|"/, '').split('.')
+      if parts.length > 1
```
Comment (coderanger) [inline lib/foodcritic/rules.rb]: Do we want to check `> 3` too?

Comment (lamont-granquist) [inline lib/foodcritic/rules.rb]: Yeah, this is just a rebase of the existing code, I want to add tests and fix that.  CBVs actually have to be exactly 3 dotted decimals and this code can most likely be simplified considerably.

Comment (coderanger) [inline lib/foodcritic/rules.rb]: Probably okay to accept one or two parts and let it extend them, or at least that is pretty common and doesn't seem like a bad idea.

Comment (lamont-granquist) [inline lib/foodcritic/rules.rb]: Okay I didn't realize 'x.y' was acceptable:

```
ERROR: Chef::Exceptions::InvalidCookbookVersion: '1' does not match 'x.y.z' or 'x.y'
```

We should just do that.

**Comment by lamont-granquist** [comment]
@coderanger @tas50 @acrmp hows this look now?

**Comment by coderanger** [comment]
I would make throw in one test using double quotes because I'm paranoid, but :+1: :shipit:

**Comment by acrmp** [comment]
A `metadata.rb` with this would also match incorrectly - don't know how common it would be in practice:

```ruby
expr = '1.0.0'
version expr
```

**Comment by lamont-granquist** [comment]
Uhm, yeah, that's actually terrible and blocker for this.

We need to only select string or double string literals here and check them.  Peeps with CI/CD style cookbook delivery systems may be calling out to random ruby libraries to set their versions.

**Comment by lamont-granquist** [comment]
Also we should probably not file_match but match with a line number...

**Comment by lamont-granquist** [comment]
@coderanger @acrmp @tas50 think this one is ready.

File: lib/foodcritic/rules.rb [hunk 2]
```diff
@@ -826,3 +826,21 @@ def invalid_name(ast)
     end
   end
 end
+
+rule 'FC061', 'Valid cookbook versions are of the form x.y or x.y.z' do
+  tags %w{metadata}
```
Comment (tas50) [inline lib/foodcritic/rules.rb]: We should also include the tag correctness here

**Comment by acrmp** [comment]
Thanks. I think FC061 would still match this case:

```ruby
version generated('like this')
```

In some rules I believe we count ancestors to avoid this kind of issue.

We could also return the matching node so that we get the correct line number rather than using `file_match`.

**Comment by lamont-granquist** [comment]
working on a fix...

**Pull request merged**